src/pypi_mapping/custom_pypi_mapping.rs

```diff
@@ -160,12 +160,17 @@ fn amend_pypi_purls_for_record(
     if record
         .package_record
         .purls
-        .iter()
-        .any(|p| p.package_type() == "pypi")
+        .as_ref()
+        .is_some_and(|vec| vec.iter().any(|p| p.package_type() == "pypi"))
     {
         return Ok(());
     }
 
+    // Initialize the vector if it's None
+    if record.package_record.purls.is_none() {
+        record.package_record.purls = Some(Vec::new());
+    }
```

Comment:
Maybe instead of first assigning a value it makes sense to construct the value in the code below and only then assign it. This would forgo the expect you have down below as well.